### PR TITLE
Misc cleanups and minor formatting

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/metar/default.nix [hunk 1]
```diff
@@ -27,5 +26,7 @@ stdenv.mkDerivation {
       more work in the area of clouds need to be done, as support for Cumulus or
       Cumulunimbus is not yet decoded.
     '';
+    license = licenses.gpl2;
+    maintainers = [ maintainers.zalakain ];
```
Comment (figsoda) [inline pkgs/applications/misc/metar/default.nix:30]:
```suggestion
    license = licenses.gpl2Plus;
    maintainers = with maintainers; [ zalakain ];
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/gkrellm/default.nix [hunk 2]
```diff
@@ -15,8 +15,6 @@
 , wrapGAppsHook
 }:
 
-with lib;
```
Comment (figsoda) [inline pkgs/applications/misc/gkrellm/default.nix:18]: there is a `optionals` below that needs to be changed to `lib.optionals` or `lib.optional`

File: pkgs/applications/misc/audio/sox/default.nix [hunk 3]
```diff
@@ -5,36 +5,34 @@
...
-, amrwb ? null
-, enableLibpulseaudio ? true
-, libpulseaudio ? null
+, amrnb
+, amrwb
+, enableLibpulseaudio ? stdenv.isLinux
+, libpulseaudio
 }:
 
-with lib;
```
Comment (figsoda) [inline pkgs/applications/misc/audio/sox/default.nix:36]: there is a `optional` below that needs to be changed to `lib.optional`

Comment (figsoda) [reply pkgs/applications/misc/audio/sox/default.nix:36]: should be the last one we missed

File: pkgs/applications/misc/loxodo/default.nix [hunk 4]
```diff
@@ -1,19 +1,17 @@
-{ lib, python27Packages, fetchgit }:
-let
-  py = python27Packages;
-  python = py.python;
-in
-py.buildPythonApplication {
+{ lib, python2, fetchgit }:
```
Comment (figsoda) [inline pkgs/applications/misc/loxodo/default.nix:1]:
```suggestion
{ lib, python2, fetchFromGitHub }:
```

File: pkgs/applications/misc/metar/default.nix [hunk 5]
```diff
@@ -2,10 +2,11 @@
 
 stdenv.mkDerivation {
   pname = "metar";
-  version = "20161013.1";
+  version = "unstable-2017-02-17";
 
-  src = fetchgit {
-    url = "https://github.com/keesL/metar.git";
+  src = fetchFromGitHub {
```
Comment (figsoda) [inline pkgs/applications/misc/metar/default.nix:7]: fetchgit above needs to be changed to fetchFromGitHub

Comment (SuperSandro2000) [reply pkgs/applications/misc/metar/default.nix:7]: Why does neither the LSP nor ``nix-env -qaP`` catch those errors?

Comment (figsoda) [reply pkgs/applications/misc/metar/default.nix:7]: I don't think rnix-lsp has any special support for nixpkgs yet
I don't know about `nix-env -qaP`, but it's probably a good idea to do a `nixpkgs-review rev HEAD`

File: pkgs/applications/misc/garmin-plugin/default.nix [hunk 6]
```diff
@@ -1,28 +1,37 @@
...
-  src = fetchurl {
-    url = "https://github.com/adiesner/GarminPlugin/archive/V${version}.tar.gz";
+  src = fetchFromGitHub {
+    owner = "adiesner";
+    repo = "GarminPlugin";
+    rev = "V${version}";
     sha256 = "15gads1fj4sj970m5960dgnhys41ksi4cm53ldkf67wn8dc9i4k0";
   };
+
   sourceRoot = "GarminPlugin-${version}/src";
```
Comment (figsoda) [inline pkgs/applications/misc/garmin-plugin/default.nix:14]:
```suggestion
  sourceRoot = "src";
```

File: pkgs/applications/misc/metar/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/metar/default.nix:7]: Yeah, I should have done that.

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 145101` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>loxodo</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>garmin-plugin</li>
    <li>metar</li>
    <li>procmail</li>
    <li>rofi-menugen</li>
    <li>stag</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**